ecs-init/docker/docker_test.go

```diff
@@ -985,3 +987,99 @@ func TestDefaultIsPathValid(t *testing.T) {
+		expectedBind                         string
+	}{
+		{
+			name:                                 "No Credentials Fetcher host from env",
+			credentialsFetcherHostFromEnv:        "",
+			credentialsFetcherHostFromConfigFile: "dummy",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock:/var/credentials-fetcher/socket/credentials_fetcher.sock",
+		},
+		{
+			name:                                 "Invalid Credentials Fetcher host from env",
+			credentialsFetcherHostFromEnv:        "invalid",
+			credentialsFetcherHostFromConfigFile: "dummy",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock:/var/credentials-fetcher/socket/credentials_fetcher.sock",
+		},
+		{
+			name:                                 "Credentials Fetcher from env, no Credentials Fetcher from config file",
+			credentialsFetcherHostFromEnv:        "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock",
+			credentialsFetcherHostFromConfigFile: "",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock:/var/credentials-fetcher/socket/credentials_fetcher.sock",
+		},
+		{
+			name:                                 "Credentials Fetcher from env, invalid Credentials Fetcher from config file",
+			credentialsFetcherHostFromEnv:        "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock",
+			credentialsFetcherHostFromConfigFile: "invalid",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock:/var/credentials-fetcher/socket/credentials_fetcher.sock",
+		},
+		{
+			name:                                 "Credentials Fetcher host from env, Credentials Fetcher from config file",
+			credentialsFetcherHostFromEnv:        "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock.1",
+			credentialsFetcherHostFromConfigFile: "unix:///var/credentials-fetcher/socket/credentials_fetcher.sock.1",
+			expectedBind:                         "/var/credentials-fetcher/socket/credentials_fetcher.sock.1:/var/credentials-fetcher/socket/credentials_fetcher.sock.1",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			os.Setenv("CREDENTIALS_FETCHER_HOST", tc.credentialsFetcherHostFromEnv)
+			defer os.Unsetenv("CREDENTIALS_FETCHER_HOST")
+
+			bind, _ := getCredentialsFetcherSocketBind()
+			assert.Equal(t, tc.expectedBind, bind)
+		})
+	}
+}
+
+func TestInstanceDomainJoined(t *testing.T) {
+	execCommand = fakeExecCommand
+	defer func() { execCommand = exec.Command }()
+
+	defer func() { execLookPath = exec.LookPath }()
+
+	execLookPath = func(name string) (string, error) {
+		return "/usr/sbin/realm", nil
+	}
+
+	out := isDomainJoined()
+	assert.True(t, out)
+}
+
+func TestInstanceDomainJoinedRealmNotFound(t *testing.T) {
+	execCommand = fakeExecCommand
+	defer func() { execCommand = exec.Command }()
+
+	defer func() { execLookPath = exec.LookPath }()
```

Comment:
super-nit: this can be defined after `execLookPath` for consistency with how you defined execCommand and its defer above.

Reply:
even better would be to have just 1 defer for both execCommand and execLookPath

Reply:
moved to common defer